src/unstract/sdk/file_storage/fs_permanent.py

```diff
@@ -0,0 +1,42 @@
+from typing import Union
+
+from fsspec import AbstractFileSystem
+
+from unstract.sdk.exceptions import FileStorageError
+from unstract.sdk.file_storage.constants import Common
+from unstract.sdk.file_storage.fs_impl import FileStorage
+from unstract.sdk.file_storage.fs_provider import FileStorageProvider
+
+
+class PermanentFileStorage(FileStorage):
+    SUPPORTED_FILE_STORAGE_TYPES = [
+        FileStorageProvider.GCS.value,
+        FileStorageProvider.S3.value,
+        FileStorageProvider.Azure.value,
+    ]
+
+    def __init__(self, provider, credentials: dict() = None):
+        if provider.value not in self.SUPPORTED_FILE_STORAGE_TYPES:
+            raise FileStorageError(
+                f"File storage provider is not supported in Permanent mode. "
+                f"Supported providers: {self.SUPPORTED_FILE_STORAGE_TYPES}"
+            )
+        if provider == FileStorageProvider.GCS:
+            super().__init__(provider, credentials)
+        else:
+            raise FileStorageError("File system not supported")
```

Comment:
How is the first if and second if block different. Could you may be add some comments?

Reply:
Second if is actually to catch non-implemented types. Changed the exception now to keep that clear..